### PR TITLE
Feature/upload journal

### PR DESCRIPTION
**Pull request by TzuHanLiang**
### DEVELOP
- 調整 APIHandler 需要回傳 IResponseData 的 code
- 在頁面中呼叫 uploadJournal API
- 在頁面中呼叫 listJournal API
- 調整 isDisplayedJournalList UI 邏輯 @Julian0701 需要幫忙檢查一下
- 調整 getJournalDetails 並在頁面中使用API回傳數據
- 調整 EventType
- 負責 Murky #572 裡面的 STATUS_CODE 在 catch fetch error 時，將 code 設定為 STATUS_CODE.INTERNAL_SERVICE_ERROR

### CHECK LIST
- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
 - new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### CODE REVIEW

File: src/components/contact_form/contact_form.tsx
Comment (Luphia) [inline src/components/contact_form/contact_form.tsx:1]: - do not use eslint-disable

File: src/components/contact_form/contact_form.tsx [hunk 1]
```diff
@@ -110,6 +111,7 @@ function ContactForm() {
     }
     if (enmailError) {
       failedProcess();
+      console.log(`enmailError(${emailCode}): `, enmailError);
```
Comment (Luphia) [inline src/components/contact_form/contact_form.tsx:114]: - require annotation for console.log

File: src/components/confirm_modal/confirm_modal.tsx [hunk 2]
```diff
@@ -84,23 +101,39 @@ const ConfirmModal = ({
...
   };
 
+  useEffect(() => {
+    if (uploadSuccess && result) {
+      modalVisibilityHandler(); // Info: (20240503 - Julian) 關閉 Modal
+      clearVoucherHandler(); // Info: (20240503 - Julian) 清空 Voucher
+      router.push(`${ISUNFA_ROUTE.ACCOUNTING}/${result.resultId}`); // Info: (20240503 - Julian) 將網址導向至 /user/accounting/[id]
+    } else {
+      // TODO: Error handling @Julian (20240510 - Tzuhan)
+      console.log(`Failed to upload journal: `, codeUploadJournal, `error: `, errorUploadJournal);
```
Comment (Luphia) [inline src/components/confirm_modal/confirm_modal.tsx:133]: - require annotation for console.log

File: src/components/camera_scanner/camera_scanner.tsx [hunk 3]
```diff
@@ -188,7 +189,7 @@ const CameraScanner = ({ isModalVisible, modalVisibilityHandler }: ICameraScanne
       setOcrResultIdHandler(resultId);
     } else {
       // eslint-disable-next-line no-console
-      console.error('Error: ', uploadError);
+      console.error('Error: ', uploadError, 'Code: ', uploadCode);
```
Comment (Luphia) [inline src/components/camera_scanner/camera_scanner.tsx:192]: - require annotation for console.error

**Review by Luphia — CHANGES_REQUESTED** [review]

File: src/components/camera_scanner/camera_scanner.tsx (hunk 3 above)
Comment (TzuHanLiang) [reply src/components/camera_scanner/camera_scanner.tsx:193]: - resolved

File: src/components/confirm_modal/confirm_modal.tsx (hunk 2 above)
Comment (TzuHanLiang) [reply src/components/confirm_modal/confirm_modal.tsx:133]: - resolved

File: src/components/contact_form/contact_form.tsx
Comment (TzuHanLiang) [reply src/components/contact_form/contact_form.tsx:1]: - resolved

File: src/components/contact_form/contact_form.tsx (hunk 1 above)
Comment (TzuHanLiang) [reply src/components/contact_form/contact_form.tsx:116]: - resolved

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**